bigquery/setup.py

```diff
@@ -20,79 +20,86 @@
 
 # Package metadata.
 
-name = 'google-cloud-bigquery'
-description = 'Google BigQuery API client library'
-version = '1.10.0'
+name = "google-cloud-bigquery"
+description = "Google BigQuery API client library"
+version = "1.10.0"
 # Should be one of:
 # 'Development Status :: 3 - Alpha'
 # 'Development Status :: 4 - Beta'
 # 'Development Status :: 5 - Production/Stable'
-release_status = 'Development Status :: 5 - Production/Stable'
+release_status = "Development Status :: 5 - Production/Stable"
 dependencies = [
-    'google-api-core >= 1.6.0, < 2.0.0dev',
-    'google-cloud-core >= 0.29.0, < 0.30dev',
-    'google-resumable-media >= 0.3.1',
+    "google-api-core >= 1.6.0, < 2.0.0dev",
+    "google-cloud-core >= 0.29.0, < 0.30dev",
+    "google-resumable-media >= 0.3.1",
 ]
 extras = {
-    'bqstorage': 'google-cloud-bigquery-storage >= 0.2.0dev1, <2.0.0dev',
-    'pandas': 'pandas>=0.17.1',
+    "bqstorage": "google-cloud-bigquery-storage[fastavro] >= 0.2.0dev1, <2.0.0dev",
+    "pandas": "pandas>=0.17.1",
     # Exclude PyArrow dependency from Windows Python 2.7.
-    'pyarrow: platform_system != "Windows" or python_version >= "3.4"':
-        'pyarrow>=0.4.1',
-    'tqdm': 'tqdm >= 4.0.0, <5.0.0dev',
-    'fastparquet': ['fastparquet', 'python-snappy'],
+    'pyarrow: platform_system != "Windows" or python_version >= "3.4"': "pyarrow>=0.4.1",
+    "fastparquet": ["fastparquet", "python-snappy"],
+    "tqdm": "tqdm >= 4.0.0, <5.0.0dev",
 }
 
+all_extras = []
+for extra in extras:
+    if isinstance(extras[extra], str):
+        all_extras.append(extras[extra])
+    else:
+        all_extras.extend(extras[extra])
+
+extras["all"] = all_extras
```

Comment:
Can't we just normalize the extras above to always be lists?

Reply:
I fear if we miss one, then we end up with a bad dependency list.

```
>>> l = []
>>> l.extend("Uh, oh!")
>>> l
['U', 'h', ',', ' ', 'o', 'h', '!']
```

That means we still would need a check for string type or something else to validate they are all lists.

Reply:
I don't what I was thinking with that comment. This is why we have tests. I assume by using the [all] extra in the nox session, we'd catch such errors. I'll simplify this on Monday as you suggest.

Reply:
Done in https://github.com/googleapis/google-cloud-python/pull/7610/commits/17e65d7ce393ff8f2e41bf1d465fcd833c6d44bc.